### ISSUE TITLE
runs for a while then the whole system freeze...

### ISSUE THREAD

**Issue by jeanrocco** [janztec/empc-arpi-linux-readonly#3]
Hello, 
I installed the script on a fresh: 
cat os-release
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

and it runs fine for a few hours, then freezes completely, no ssh access, no console access, it gives me input-output errors...
Any clues ? what should I show you to find out what the problem is...
Very impressed by you scripting level, if it works it will cure the PI un-reliability problem ... Thank's
jeanrocco

**Comment by philippejadin** [comment]
You are probably runing out of (ram-) disk space. Try to check the available disk space while running before the crash and it will give you a clue on that. Afaict, there is 256mb of ram used for that purpose. If your application writes a lot of log files, it might fill that space. You could also redirect your logs to dev/null or something.
hth

**Comment by janztec-ama** [comment]
Hi jeanrocco,

can you record the output of "df" for example every hour?
**pi@raspberrypi:/home/pi $ sudo df**

which kernel version are you using?
**pi@raspberrypi:/home/pi $ sudo uname -a**

Also we have updated file README.md with basic information on how to optimize logging.
ama

**Comment by jeanrocco** [comment]
Hello Philippe,
Thank’s for your answer, you  give me hope since, as you say, it was working flawlessly before, so it just a matter of tweaking it correctly. By the way I just now received an e-mail from janztec-ama giving me hints on how to pinpoint the problem. If you fell like it I could come back to you with my results. Janztec-ama also tells me they updated the file README.md with basic information on how to optimize logging. I will have a look at that. 
I was working on a hardware solution that you might see at https://github.com/jeanrocco/Gracefull-PI-shutdown-and-auto-restart <https://github.com/jeanrocco/Gracefull-PI-shutdown-and-auto-restart> , when I stumbled on this overlayfs twist. We agree that the root read only approach is much much more simpler …, an probably more reliable too. It happens that hardware is easier for me than soft, so … :-) ! 
A la prochaine,
jrb.

**Comment by philippejadin** [comment]
Regarding ups for pi, you might find this interesting : https://www.modmypi.com/raspberry-pi/power-1051/ups-boards-1051/ups-pico

**Comment by jeanrocco** [comment]
Hello Janztec-ama, thank's for your answer.
I will reinstall your script, and record "df" ... as soon as I can ...
I was running "home assistant" when the system froze, it is known to log a lot. Raspbian would restart fine when I rebooted but crashed a few hours later. This time I will only use Raspbian as is. 
The command:
"sudo uname -a" gives:
**Linux Hass_PI3B 4.14.70-v7+ #1144 SMP Tue Sep 18 17:34:46 BST 2018 armv7l GNU/Linux**
As Philippejadin says, I'm probably running out of rams, I will try to run **iotop -oaP** to look at disk activity.
Thank's this will be great if it works ...

**Comment by jeanrocco** [comment]
Hello ama, 
ok I ran your install-experimental.sh on a fresh: 
Linux rpi3b_Hass 4.14.71-v7+ #1145 SMP Fri Sep 21 15:38:35 BST 2018 armv7l GNU/Linux
and it works flawlessly, this is great !!! 
here is my df output:
Filesystem     1K-blocks    Used Available Use% Mounted on
udev              490052       0    490052   0% /dev
tmpfs             100020    3060     96960   4% /run
/dev/mmcblk0p2  15202996 3437184  11092140  24% /ro
overlay-rw        262144  130248    131896  50% /rw
overlay-root      262144  130248    131896  50% /
tmpfs             500088       0    500088   0% /dev/shm
tmpfs               5120       4      5116   1% /run/lock
tmpfs             500088       0    500088   0% /sys/fs/cgroup
/dev/mmcblk0p1     44144   33357     10787  76% /boot
tmpfs             100016       0    100016   0% /run/user/1000

I can see that overlay-rw and overlay-root grew to 50% when I ran vncserver and chromium browser, but are stable at that point. 
I initially tried to disable swap, and all the actions you recommended to reduce logs, but the install would not work then, and was giving me write errors. So I guess these actions should be performed after you first run the install script one time, then reboot,  and then set the system back to rw, reboot and apply the actions... 

I will now install "Home Asistant" and try to find out how it affects the system. It will probably fill out overlay-rw and overlay-root... with logs, I can report my findings back to you if you are interested, but I think you (or I) could close this issue since your script works very well and what's left is only tweaking it further for my specific needs ! 

Once again thank's for this amazing script.
jeanrocco
